### PR TITLE
WIP fix for decrypting password fields in workflow creation

### PR DESCRIPTION
**Pull request by DavidjohnBlodgett**
resolves RackHD/RackHD#417

@RackHD/corecommitters

@keedya

This resolves the issue Andre was seeing where the script file was being rendered with an encrypted password.

However, I have locally been hitting an issue where, when I try to submit the graph to run against a node, I hit a `JSON schema validation` error, shown below.

```{
    "UUID": "80c7e41c-d957-4c92-bc69-c33f0a44f111",
    "message": "Task.Set.BMC.Credentials: JSON schema validation failed - data.password should NOT be longer than 20 characters, data.password should match pattern &#34;^[ -~]{1,20}$&#34;",
    "status": "400"
}```

I have an open question out to Andre about what commit he's running... I assume, since he did not hit this error, it is older then around Aug... ( when this schema seems to have showed up here: https://github.com/RackHD/on-tasks/blob/5ffd1780a1bf8783152a8f04202b49a0b4fbb1ac/lib/task-data/schemas/set-bmc-credentials.json )

Currently still working on a fix to decrypt the password field at this validation stage, but what is here should unblock Andre's work.

### CODE REVIEW

File: lib/services/workflow-api-service.js [hunk 1]
```diff
@@ -120,6 +123,7 @@ function workflowApiServiceFactory(
 
     WorkflowApiService.prototype.createGraph = function(definition, options, context, domain) {
         domain = domain || Constants.DefaultTaskDomain;
+        sanitizer.decrypt(options);
```
Comment (DavidjohnBlodgett) [inline lib/services/workflow-api-service.js]: oops, I meant to take this out~ please ignore for now

**Comment by rackhdmirror** [comment]
Build finished.

File: lib/services/workflow-api-service.js [hunk 2]
```diff
@@ -93,6 +95,7 @@ function workflowApiServiceFactory(
             });
         })
         .then(function(graph) {
+            sanitizer.decrypt(graph);
```
Comment (benbp) [inline lib/services/workflow-api-service.js]: I'm not quite sure what impact this has other than sending unencrypted values back to the client?

File: lib/services/workflow-api-service.js [hunk 3]
```diff
@@ -33,7 +34,8 @@ function workflowApiServiceFactory(
     Constants,
     _,
     env,
-    lookupService
+    lookupService,
```
Comment (benbp) [inline lib/services/workflow-api-service.js]: Unused?

Comment (DavidjohnBlodgett) [inline lib/services/workflow-api-service.js]: not sure, it was there when I made the edits~ would you like me to remove it?

File: lib/services/workflow-api-service.js (hunk 2 above)
Comment (DavidjohnBlodgett) [inline lib/services/workflow-api-service.js]: so, not really sure I follow... if your worried this will not fix the issue, I have tested it locally.

**Comment by DavidjohnBlodgett** [comment]
So I discovered that my method for testing the resolution was flawed, @benbp 's comments about this only effecting the client http response (in effect nullifying my earlier redaction work) was correct, please standby until I can get the correct fix pushed.

**Comment by rackhdmirror** [comment]
Build finished.

**Comment by rackhdmirror** [comment]
Build finished.

**Comment by DavidjohnBlodgett** [comment]
new PR with good fix: https://github.com/RackHD/on-core/pull/220 closing this PR.

**Pull request closed without merging**